vnext/ReactWindowsCore/Modules/BatchingUIManagerModule.cpp

```diff
@@ -0,0 +1,245 @@
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    auto arg1 = jsArgAsArray(args, 1);
+    static_cast<IUIManager*>(manager.get())->setChildren(jsArgAsInt(args, 0), std::move(arg1));
+  });
+}
+
+void BatchingUIManager::updateView(folly::dynamic&& args)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    auto arg2 = jsArgAsDynamic(args, 2);
+    static_cast<IUIManager*>(manager.get())->updateView(jsArgAsInt(args, 0), jsArgAsString(args, 1), std::move(arg2));
+  });
+}
+
+void BatchingUIManager::removeSubviewsFromContainerWithID(folly::dynamic&& args)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    static_cast<IUIManager*>(manager.get())->removeSubviewsFromContainerWithID(jsArgAsInt(args, 0));
+  });
+}
+
+void BatchingUIManager::manageChildren(folly::dynamic&& args)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    auto arg1 = jsArgAsDynamic(args, 1);
+    auto arg2 = jsArgAsDynamic(args, 2);
+    auto arg3 = jsArgAsDynamic(args, 3);
+    auto arg4 = jsArgAsDynamic(args, 4);
+    auto arg5 = jsArgAsDynamic(args, 5);
+    static_cast<IUIManager*>(manager.get())->manageChildren(jsArgAsInt(args, 0), arg1, arg2, arg3, arg4, arg5);
+  });
+}
+
+void BatchingUIManager::dispatchViewManagerCommand(folly::dynamic&& args)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    auto arg2 = jsArgAsDynamic(args, 2);
+    static_cast<IUIManager*>(manager.get())->dispatchViewManagerCommand(jsArgAsInt(args, 0), jsArgAsInt(args, 1), std::move(arg2));
+  });
+}
+
+void BatchingUIManager::replaceExistingNonRootView(folly::dynamic&& args)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args]()
+  {
+    static_cast<IUIManager*>(manager.get())->replaceExistingNonRootView(jsArgAsInt(args, 0), jsArgAsInt(args, 1));
+  });
+}
+
+void BatchingUIManager::measure(folly::dynamic&& args, facebook::xplat::module::CxxModule::Callback cb)
+{
+  auto manager = shared_from_this();
+  m_queue->write([manager, args, cb]()
+  {
+    static_cast<IUIManager*>(manager.get())->measure(jsArgAsInt(args, 0), cb);
+  });
+}
+
+
+BatchingUIManagerModule::BatchingUIManagerModule(std::shared_ptr<IUIManager>&& manager)
+	: m_manager(std::move(manager))
+{
+}
+
+std::string BatchingUIManagerModule::getName()
+{
+	return "UIManager";
+}
+
+std::map<std::string, folly::dynamic> BatchingUIManagerModule::getConstants()
+{
+	std::map<std::string, folly::dynamic> constants {};
+
+	m_manager->populateViewManagerConstants(constants);
+
+	return constants;
+}
+
+std::vector<facebook::xplat::module::CxxModule::Method> BatchingUIManagerModule::getMethods()
+{
+  auto manager = std::static_pointer_cast<BatchingUIManager>(m_manager);
+  return
+  {
+    Method("getConstantsForViewManager", [manager](dynamic argsJson) -> dynamic
+    {
+      auto args = folly::parseJson(argsJson.asString());
```

Comment:
Since this is duplicating logic in UIManagerModule, recent change I made there for JSI is needed here too

Reply:
Thanks. Ported.